doc/metadata_routing.rst

```diff
@@ -179,42 +197,39 @@ consumers. In this example, we pass ``scoring_weight`` to the scorer, and
+scorers, or splitters) and don't use the metadata themselves are not consumers.
+(Meta-)Estimators which route metadata to other objects are :term:`routers <router>`.
+A(n) (meta-)estimator can be a :term:`consumer` and a :term:`router` at the same time.
+(Meta-)Estimators and splitters expose a `set_{method}_request` method for each method
+which accepts at least one metadata. For instance, if an estimator supports
+``sample_weight`` in ``fit`` and ``score``, it exposes
 ``estimator.set_fit_request(sample_weight=value)`` and
 ``estimator.set_score_request(sample_weight=value)``. Here ``value`` can be:
 
-- ``True``: method requests a ``sample_weight``. This means if the metadata is
-  provided, it will be used, otherwise no error is raised.
+- ``True``: method requests a ``sample_weight``. This means if the metadata is provided,
+  it will be used, otherwise no error is raised.
 - ``False``: method does not request a ``sample_weight``.
-- ``None``: router will raise an error if ``sample_weight`` is passed. This is
-  in almost all cases the default value when an object is instantiated and
-  ensures the user sets the metadata requests explicitly when a metadata is
-  passed. The only exception are ``Group*Fold`` splitters.
-- ``"param_name"``: if this estimator is used in a meta-estimator, the
-  meta-estimator should forward ``"param_name"`` as ``sample_weight`` to this
-  estimator. This means the mapping between the metadata required by the
-  object, e.g. ``sample_weight`` and what is provided by the user, e.g.
-  ``my_weights`` is done at the router level, and not by the object, e.g.
-  estimator, itself.
-
-Metadata are requested in the same way for scorers using ``set_score_request``.
-
-If a metadata, e.g. ``sample_weight``, is passed by the user, the metadata
-request for all objects which potentially can consume ``sample_weight`` should
-be set by the user, otherwise an error is raised by the router object. For
-example, the following code raises an error, since it hasn't been explicitly
-specified whether ``sample_weight`` should be passed to the estimator's scorer
-or not::
+- ``None``: router will raise an error if ``sample_weight`` is passed. This is in almost
+  all cases the default value when an object is instantiated and ensures the user sets
+  the metadata requests explicitly when a metadata is passed. The only exception are
+  ``Group*Fold`` splitters.
+- ``"param_name"``: alias for ``sample_weight`` if we want to pass different weights to
+  different consumers. If aliasing is used the meta-estimator should not forward
+  ``"param_name"`` to the consumer, but ``sample_weight`` instead, because the consumer
+  will expect a param called ``sample_weight``. This means the mapping between the
+  metadata required by the object, e.g. ``sample_weight`` and the variable name provided
+  by the user, e.g. ``my_weights`` is done at the router level, and not by the consuming
+  object itself.
```

Comment:
And I would suggest to remove the part (starting from "_If aliasing ...."_) from here and add it to the example (examples/miscellaneous/plot_metadata_routing.py). The reason for this being is that people who only want to use metadata routing don't need to be burdened with this information.

I'd do that in PR #27357. Is this okay?

Reply:
No I rather keep it here for the same of completeness of the API section.